src/ImageSharp/Formats/Jpg/Components/Block8x8F.cs

```diff
@@ -15,6 +15,11 @@ namespace ImageSharp.Formats
     /// </summary>
     internal partial struct Block8x8F
     {
+        // Most of the static methods of this struct are instance methods by actual semantics: they use Block8x8F* as their first parameter.
```

Comment:
Slow:
```
public unsafe float this[int idx]
{
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    get
    {
        fixed (Block8x8F* p = &this)
        {
            float* fp = (float*)p;
            return fp[idx];
        }
    }
}
```

Fast:
```
[MethodImpl(MethodImplOptions.AggressiveInlining)]
public static unsafe float GetScalarAt(Block8x8F* blockPtr, int idx)
{
    float* fp = (float*)blockPtr;
    return fp[idx];
}
```

There's no need to fix the block if it's on the stack, or it's member of an already-fixed heap object. Unfortunately the only way to "express" it in C# is to have a static method that takes `Block8x8F*` :(